eng/pipelines/templates/pipeline.yml

```diff
@@ -1,6 +1,22 @@
 jobs:
+- job: Initialize_Build_SemanticVersion
+  timeoutInMinutes: 2
+  pool:
+    name: VSEng-MicroBuildVS2019
+    demands:
+      - DotNetFramework
+      - msbuild
+
+  steps:
+  - template: Initialize_Build_SemanticVersion.yml
+
 - job: Initialize_Build
+  dependsOn: Initialize_Build_SemanticVersion
   timeoutInMinutes: 10
+  variables:
+    SemanticVersion: $[dependencies.Initialize_Build_SemanticVersion.outputs['setsemanticversion.SemanticVersion']]
+    BuildRevision: $[counter(format('{0}.{1}', variables['SemanticVersion'], variables['build.definitionname']), 7130)]
```

Comment:
>  This approach ensures that for every CI build, version number of binaries is higher than previous version of the product. After we merge this PR into dev and cherry-pick these commits onto our servicing branches then I will create a follow up PR to change the seed back to 0 so that CI revision number starts from 0 for any change in major, minor or patch versions.

What's the plan for old releases if we have to service them?

Do we cherry-pick this PR's commit, but not the new PR's commit which changes the default to zero?

Reply:
One option is to cherry-pick this commit only on the need basis. second option is to cherry-pick this commit proactively into all the service branches after merging into `dev`. I lean towards second option so that we don't have any last-minute surprises. I think, we must cherry-pick other dependent commits such as https://github.com/NuGet/NuGet.Client/pull/3883 and https://github.com/NuGet/NuGet.Client/pull/3890 (may be but not sure)